### PR TITLE
[FLINK-6749] [table] Table API / SQL Docs: SQL Page

### PR DESCRIPTION
**Pull request by haohui**
This first iteration of the documentation. Comments are appreciated.

### CODE REVIEW

File: docs/dev/table/sql.md [hunk 1]
```diff
@@ -22,20 +22,22 @@ specific language governing permissions and limitations
 under the License.
 -->
 
-SQL queries are specified using the `sql()` method of the `TableEnvironment`. The method returns the result of the SQL query as a `Table` which can be converted into a `DataSet` or `DataStream`, used in subsequent Table API queries, or written to a `TableSink` (see [Writing Tables to External Sinks](#writing-tables-to-external-sinks)). SQL and Table API queries can seamlessly mixed and are holistically optimized and translated into a single DataStream or DataSet program.
+Flink supports specifying DataStream or DataSet programs with SQL queries using the `sql()` method of the `TableEnvironment`. The method returns the result of the SQL query as a `Table`. A `Table` can be used in the subsequent SQL / Table API queries, be converted into a `DataSet` or `DataStream`, used in subsequent Table API queries or written to a `TableSink` (see [Writing Tables to External Sinks](common.html#emit-to-a-tablesink)). SQL and Table API queries can seamlessly mixed and are holistically optimized and translated into a single program.
```
Comment (alpinegizmo) [inline docs/dev/table/sql.md]: A `Table` can be used in subsequent SQL / Table API queries, be converted into a `DataSet` or `DataStream`, or written to a `TableSink` ...

_(drop "the" and the redundant "used in subsequent Table API queries")_

File: docs/dev/table/sql.md [hunk 2]
```diff
@@ -22,20 +22,22 @@ specific language governing permissions and limitations
...
+To access the data in the SQL queries, users must register data sources, including `Table`, `DataSet`, `DataStream` or external `TableSource`, in the `TableEnvironment` (see [Registering Tables](common.html#register-a-table-in-the-catalog)). Alternatively, users can also register external catalogs in the `TableEnvironment` to specify the location of the data sources.
 
-*Note: Flink's SQL support is not feature complete, yet. Queries that include unsupported SQL features will cause a `TableException`. The limitations of SQL on batch and streaming tables are listed in the following sections.*
+For convenience `Table.toString()` will automatically register an unique table name under the `Table`'s `TableEnvironment` and return the table name. So it allows to call SQL directly on tables in a string concatenation (see examples below).
```
Comment (alpinegizmo) [inline docs/dev/table/sql.md]: ... a unique table name ...

Comment (alpinegizmo) [inline docs/dev/table/sql.md]: This allows SQL to be called directly on tables in a string concatenation (see examples below).

File: docs/dev/table/sql.md [hunk 3]
```diff
@@ -22,20 +22,22 @@ specific language governing permissions and limitations
...
 
-**TODO: Rework intro. Move some parts below. **
+*Note: Flink's SQL support is not feature complete, yet. Queries that include unsupported SQL features will cause a `TableException`. The limitations of SQL on batch and streaming tables are listed in the following sections.*
```
Comment (alpinegizmo) [inline docs/dev/table/sql.md]: Flink's SQL support is not yet feature complete.

File: docs/dev/table/sql.md [hunk 4]
```diff
@@ -22,20 +22,22 @@ specific language governing permissions and limitations
...
 
 * This will be replaced by the TOC
 {:toc}
 
 Specifying a Query
 ---------------
 
+Here are a few examples on how to specify a DataStream / DataSet program using SQL:
```
Comment (alpinegizmo) [inline docs/dev/table/sql.md]: examples of how to

File: docs/dev/table/sql.md [hunk 5]
```diff
@@ -82,14 +84,12 @@ val result2 = tableEnv.sql(
...
 
-**TODO: Add some intro.**
-
 {% top %}
 
 Supported Syntax
 ----------------
 
-Flink uses [Apache Calcite](https://calcite.apache.org/docs/reference.html) for SQL parsing. Currently, Flink SQL only supports query-related SQL syntax and only a subset of the comprehensive SQL standard. The following BNF-grammar describes the supported SQL features:
+Flink parses SQL using [Apache Calcite](https://calcite.apache.org/docs/reference.html). Flink supports the standard ANSI SQL but it provides no supports for DML and DDL. The following BNF-grammar describes the supported SQL features:
```
Comment (alpinegizmo) [inline docs/dev/table/sql.md]: Flink supports standard ANSI SQL, but it provides no support for DML or DDL.

File: docs/dev/table/sql.md [hunk 6]
```diff
@@ -22,20 +22,22 @@ specific language governing permissions and limitations
...
 
-A `Table`, `DataSet`, `DataStream`, or external `TableSource` must be registered in the `TableEnvironment` in order to be accessible by a SQL query (see [Registering Tables](#registering-tables)). For convenience `Table.toString()` will automatically register an unique table name under the `Table`'s `TableEnvironment` and return the table name. So it allows to call SQL directly on tables in a string concatenation (see examples below).
+To access the data in the SQL queries, users must register data sources, including `Table`, `DataSet`, `DataStream` or external `TableSource`, in the `TableEnvironment` (see [Registering Tables](common.html#register-a-table-in-the-catalog)). Alternatively, users can also register external catalogs in the `TableEnvironment` to specify the location of the data sources.
```
Comment (alpinegizmo) [inline docs/dev/table/sql.md]: This reads a bit awkwardly. How about this?

Before using data in a SQL query, the data source(s) must first be registered in the `TableEnvironment` (see [Registering Tables](common.html#register-a-table-in-the-catalog)). Possible data sources include `Table`s, `DataSet`s, `DataStream`s, and external `TableSource`s. Alternatively, ...

File: docs/dev/table/sql.md [hunk 7]
```diff
@@ -340,18 +567,14 @@ val result4 = tableEnv.sql(
 
 {% top %}
 
-### Limitations
-
-**TODO: Integrate this with the examples**
-
 #### Batch
```
Comment (fhueske) [inline docs/dev/table/sql.md]: I would remove the `Batch` and `Streaming` limitations sections. 
Instead, I would integrate the relevant information with the `Operations` sections, e.g., add a comments to UNNEST that WITH ORDINALITY is not supported, etc.

**Comment by fhueske** [comment]
I've seen several questions on the mailing list and Stack Overflow related to breaking changes in the 1.3 release (mostly about the time indicator). That's why I would like to update the documentation ASAP.

I will integrate my comments and merge this PR.

**Comment by fhueske** [comment]
Hi @haohui, I merged the PR to the `tableDocs` branch.
Can you close it? Thanks!

**Pull request closed without merging**